inst/resources/bs4_book/bs4_book.js

```diff
@@ -0,0 +1,132 @@
+  if (toMark) {
+    mark.mark(toMark, {
+      accuracy: {
+        value: "complementary",
+        limiters: [",", ".", ":", "/"],
+      }
+    });
+  }
+
+  // Activate popovers
+  $('[data-toggle="popover"]').popover({
+    container: 'body',
+    html: true,
+    trigger: 'focus',
+    placement: "top",
+    sanitize: false,
+  });
+  $('[data-toggle="tooltip"]').tooltip();
+})
+
+// Search ----------------------------------------------------------------------
+
+var fuse;
+
+$(function () {
+  // Initialise search index on focus
+  $("#search").focus(async function(e) {
+    if (fuse) {
+      return;
+    }
+
+    $(e.target).addClass("loading");
+
+    var response = await fetch('search.json');
+    var data = await response.json();
+
+    var options = {
+      keys: ["heading", "text", "code"],
+      ignoreLocation: true,
+      threshold: 0.1,
+      includeMatches: true,
+      includeScore: true,
+    };
+    fuse = new Fuse(data, options);
+
+    $(e.target).removeClass("loading");
+  });
+
+  // Use algolia autocomplete
+  var options = {
+    autoselect: true,
+    debug: true,
+    hint: false,
+    minLength: 2,
+  };
+
+  $("#search").autocomplete(options, [
+    {
+      name: "content",
+      source: searchFuse,
+      templates: {
+        suggestion: (s) => {
+          if (s.chapter == s.heading) {
+            return `${s.chapter}`;
+          } else {
+            return `${s.chapter} /<br> ${s.heading}`;
+          }
+        },
+      },
+    },
+  ]).on('autocomplete:selected', function(event, s) {
+    window.location.href = s.path + "?q=" + q + "#" + s.id;
+  });
+});
+
+var q;
+async function searchFuse(query, callback) {
+  await fuse;
+
+  var items;
+  if (!fuse) {
+    items = [];
+  } else {
+    q = query;
+    var results = fuse.search(query, { limit: 20 });
+    items = results
+      .filter((x) => x.score <= 0.75)
+      .map((x) => x.item);
+  }
+
+  console.log(results);
+  callback(items);
+}
+
+// Copy to clipboard -----------------------------------------------------------
+
+function changeTooltipMessage(element, msg) {
+  var tooltipOriginalTitle=element.getAttribute('data-original-title');
+  element.setAttribute('data-original-title', msg);
+  $(element).tooltip('show');
+  element.setAttribute('data-original-title', tooltipOriginalTitle);
+}
+
+$(document).ready(function() {
+  if(ClipboardJS.isSupported()) {
+    // Insert copy buttons
+    var copyButton = "<div class='copy'><button type='button' class='btn btn-outline-primary btn-copy' title='Copy to clipboard' aria-label='Copy' data-toggle='popover' data-placement='top' data-trigger='hover'>Copy</button></div>";
+    $(copyButton).prependTo("pre");
+    // Initialize tooltips:
+    $('.btn-copy').tooltip({container: 'body', boundary: 'window'});
+
+    // Initialize clipboard:
+    var clipboard = new ClipboardJS('.btn-copy', {
+      text: function(trigger) {
+        return trigger.parentNode.parentNode.textContent;
```

Comment:
I think you need this here to solve https://twitter.com/TazPoltorak/status/1321677145404542977?s=20
```suggestion
        return trigger.parentNode.nextSibling.textContent;
```
The chunk code content is the node after the copy button. If you select the all div, you get the 'Copy' text you put in the copy button. If you use an icon like in current bookdown, that would not be an issue..

Reply:
Thanks! I was just about to look into this.

Reply:
I was curious why it worked in `gitbook()` but not in `bs4_book()` so I had a quick look in the morning 😉